ext-plugin/plugin.ts

```diff
@@ -0,0 +1,101 @@
+"use strict";
+import * as cp from "child_process";
+
+function main(arg:string) {
+    const txt:string = arg;
+    const txtArr:string[]|null = modifyCommentedText(txt);
+    if (txtArr != null) {
+        const textToBeSearched:string = txtArr[0];
+        const commentBegin:string  = txtArr[1];
+        const commentEnd:string = txtArr[2];
+
+        spawnChild(textToBeSearched, function(myArr:string[]) {
+            howdoiResult(myArr,txt, commentBegin, commentEnd);
+        });
+    }   
+}
+
+async function spawnChild(command:string, callback:any) {
+    const updatedCommand = howdoiPrefix(command);
+    const process = await cp.spawn("howdoi", [updatedCommand, '-n 3']);
+    let result:string[] = [];
+    process.stdout.on("data", (data:any) => {
+        result.push(String(data));
+    });
+
+    process.stderr.on("data", (data:any) => {
+        console.log(`stderr: ${data}`);
+    });
+    
+    process.on('error', (error:any) => {
+        console.log(`error: ${error.message}`);
+    });
+    
+    process.on("close", (code:any) => {
+        console.log(`child process exited with code ${code}`);
+        callback(result);
+    });
+}
+
+// removes howdoi from command
+function howdoiPrefix(command:string) {
+    const prefix = "howdoi";
+    
+    if (command.includes(prefix)) {
+        const newCommand = command.replace(prefix,'');
+        return newCommand;
+    }
+    else {
+        return command;
+    }
+}
+
+//  only for single line comments
+function modifyCommentedText(textToBeModified:string) {
+    const regexBegins:RegExp =  /^[!@#<>/\$%\^\&*\)\(+=._-]+/;
+    const regexEnds:RegExp = /[!@#<>/\$%\^\&*\)\(+=._-]+$/;
+    let commentBegin:string;
+    let commentEnd:string;	
+        
+    if (textToBeModified.match(regexBegins) && textToBeModified.match(regexEnds)){
+        commentBegin = textToBeModified.match(regexBegins)!.join();
+        commentEnd = textToBeModified.match(regexEnds)!.join();
+        textToBeModified = textToBeModified.replace(regexBegins, '');
+        textToBeModified = textToBeModified.replace(regexEnds, '');
+        let result:string[]  = [textToBeModified, commentBegin, commentEnd];
+        return result;
+    }
+    else if(textToBeModified.match(regexEnds)){
+        commentEnd = textToBeModified.match(regexEnds)!.join();
+        textToBeModified = textToBeModified.replace(regexEnds, '');
+        let result:string[] = [textToBeModified,'',commentEnd];
+        return result;
+    }
+    else if(textToBeModified.match(regexBegins)){
+        commentBegin = textToBeModified.match(regexBegins)!.join();
+        textToBeModified = textToBeModified.replace(regexBegins, '');
+        let result:string[] = [textToBeModified, commentBegin, ''];
+        return result;
+    }
+    else {
+        return null;
+    }
+}
+
+function spliceArr(obj:string[], commentBegin:string, commentEnd:string) {
```

Comment:
It's hard to read exactly what this function is doing. It might be because your result structure looks like `[[answer1, link1],[answer2, link2],...]` but `obj` and `newArr` is also not descriptive so it's difficult to wrap my head around this. Some high level suggestions:

What if the underlying return type looked like:
```
interface HowdoiResult {
   answer: string
   link: string
}
```
Then you could pass around HowdoiResult[] as a return type and reference it will result.answer and result.link.

Also, if you added typescript return types to all your functions `modifyCommentedText`, `howdoiResult`, this one, etc then it will be easier to read and in your editor you'll know exactly what is being returned as you write the code.

Finally, spliceArr could have a more descriptive name. `addCommentToResult`? `blockCommentAnswer`?